pageserver/src/layered_repository.rs

```diff
@@ -1636,30 +1636,40 @@ impl LayeredTimeline {
     ///
     pub fn check_checkpoint_distance(self: &Arc<LayeredTimeline>) -> Result<()> {
         let last_lsn = self.get_last_record_lsn();
+        let layers = self.layers.read().unwrap();
+        if let Some(open_layer) = &layers.open_layer {
+            let open_layer_size = open_layer.size()?;
+            drop(layers);
+            let distance = last_lsn.widening_sub(self.last_freeze_at.load());
+            if distance >= self.get_checkpoint_distance().into()
+                || open_layer_size > self.get_checkpoint_distance()
+            {
```

Comment:
This part of this PR looks ok to me. A comment would be nice though.

Reply:
I wonder do we really need to check LSN range at all?
May be just leave the second check?
Out goal is to limit file size, not to cover soe particular LSN range, isn't it?

Reply:
> I wonder do we really need to check LSN range at all? May be just leave the second check? Out goal is to limit file size, not to cover some particular LSN range, isn't it?

We care about both. The distance between last_record_lsn and disk_consistent_lsn determines how much WAL the safekeepers need to store.